### PR TITLE
Add tracking code install test to no data tab

### PR DESCRIPTION
**Pull request by snake14**
### Description:

This depends on the core PR: [21148](https://github.com/matomo-org/matomo/pull/21148)
We're adding the tracking code install check to the no data tab for MTM.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: lang/en.json
Comment (sgiehl) [inline lang/en.json]: Looking at the other code it seems the sentence is follow by `Container Dashboard`, which is currently added by using translations for those single strings. It might be better to include both words here directly. Even though the words might need to be translated here as well, simply appending them might be incorrect in various languages. als the `the` in the end would be hard to translate for translators without knowing what follows...

Comment (snake14) [reply lang/en.json]: Thank you @sgiehl . I thinking of maintainability if the Container Dashboard name was ever changed, but you make a good point about sentence structure in other languages and context. I'll add `Container Dashboard` into the translation string.

**Review by AltamashShaikh — APPROVED** [review]
@snake14 I tested and changes looks good, I notice that if my measurable has no URL added it tries to open the Matomo instance any says tracking code not installed error. Do you think we should rather alert the user to map the URL ?

Also check with Javi/Stan if they think the note should be indented a bit to right like below 
Screenshot from 2023-08-23 07-56-52

**Comment by snake14** [comment]
> @snake14 I tested and changes looks good, I notice that if my measurable has no URL added it tries to open the Matomo instance any says tracking code not installed error. Do you think we should rather alert the user to map the URL ?
> 
> Also check with Javi/Stan if they think the note should be indented a bit to right like below

Thank you @AltamashShaikh . I'll check with Product about the desired behaviour when a URL hasn't been configured for a site. I checked about the alignment of the success message and they said it's correct how it is.

**Comment by michalkleiner** [comment]
Tested the core part and that works as expected. Looked the MTM PR here and when there's no container, the logic seems to be broken. It does say there's no published version, but the text for step 1 is missing and the installation test probably shouldn't be available at that point. Good think is that it still successfully fails.

Screenshot 2023-09-06 at 5 55 58 PM

**Comment by snake14** [comment]
> Tested the core part and that works as expected. Looked the MTM PR here and when there's no container, the logic seems to be broken. It does say there's no published version, but the text for step 1 is missing and the installation test probably shouldn't be available at that point. Good think is that it still successfully fails.

Thank you @michalkleiner . I thought I had that working, but it might have been changed while I was making revisions. I'll look into it.

**Review by AltamashShaikh — CHANGES_REQUESTED** [review]
@snake14 I tested and the changes look good, now since we have made the URL configurable. user can enter anything and I see we do have a check for valid URL, but if we add a invalid URL we get a error but loading icon keeps showing up with the input disabled.

Screenshot from 2023-09-07 08-02-06

**Comment by snake14** [comment]
> @snake14 I tested and the changes look good, now since we have made the URL configurable. user can enter anything and I see we do have a check for valid URL, but if we add a invalid URL we get a error but loading icon keeps showing up with the input disabled.

That's a good catch @AltamashShaikh . I believe that core team added that validation before merging the core changes. I wonder if a core issue should be created as that issue can't be fixed in the plugin.

**Comment by AltamashShaikh** [comment]
> > @snake14 I tested and the changes look good, now since we have made the URL configurable. user can enter anything and I see we do have a check for valid URL, but if we add a invalid URL we get a error but loading icon keeps showing up with the input disabled.
> 
> That's a good catch @AltamashShaikh . I believe that core team added that validation before merging the core changes. I wonder if a core issue should be created as that issue can't be fixed in the plugin.

@snake14 Yes create a core issue to fix it, and this will be fixed automatically :+1:

**Comment by sgiehl** [comment]
@AltamashShaikh I've created a PR in core to fix the handling of invalid URLs. 
Is this TagManager PR ready to merge? I'd like to release a new RC of Matomo today, and might be good to include this changes if possible.

**Comment by AltamashShaikh** [comment]
@sgiehl It good to be merged, will merge it

**Review by AltamashShaikh — APPROVED** [review]

**Pull request merged**
